src/transpiling/mod.rs

```diff
@@ -266,13 +268,50 @@ impl ParsedSource {
       // we need the comments to be mutable, so make it single threaded
       inner.comments.into_single_threaded(),
       &inner.globals,
-      transpile_options,
+      &resolve_transpile_options(inner.media_type, transpile_options),
       emit_options,
       &inner.diagnostics,
     ))
   }
 }
 
+fn resolve_transpile_options(
+  media_type: MediaType,
+  options: &TranspileOptions,
+) -> Cow<TranspileOptions> {
+  if options.transform_jsx {
+    let allows_jsx = match media_type {
+      MediaType::JavaScript
+      | MediaType::Mjs
+      | MediaType::Cjs
```

Comment:
We don't allow jsx in these, right?

Reply:
Just tried it out and you're right. I'll change it.